flink-runtime/src/main/java/org/apache/flink/runtime/deployment/ConsumedSubpartitionContext.java

```diff
@@ -113,23 +111,23 @@ public IndexRange getConsumedSubpartitionRange(int shuffleDescriptorIndex) {
      *
      * @param consumedSubpartitionGroups a mapping of consumed partition index ranges to
      *     subpartition ranges.
-     * @param consumedResultPartitions an iterator of {@link IntermediateResultPartitionID} for the
-     *     consumed result partitions.
-     * @param partitions all partition ids of consumed {@link IntermediateResult}.
+     * @param partitionIdToShuffleDescriptorIndexMap a map that associates each {@link
+     *     IntermediateResultPartitionID} with its corresponding shuffle descriptor index.
+     * @param partitionIdRetriever a function that retrieves the {@link
+     *     IntermediateResultPartitionID} for a given index.
      * @return a {@link ConsumedSubpartitionContext} instance constructed from the input parameters.
      */
     public static ConsumedSubpartitionContext buildConsumedSubpartitionContext(
             Map<IndexRange, IndexRange> consumedSubpartitionGroups,
-            Iterator<IntermediateResultPartitionID> consumedResultPartitions,
-            IntermediateResultPartitionID[] partitions) {
-        Map<IntermediateResultPartitionID, Integer> partitionIdToShuffleDescriptorIndexMap =
-                new HashMap<>();
-        while (consumedResultPartitions.hasNext()) {
-            IntermediateResultPartitionID partitionId = consumedResultPartitions.next();
-            partitionIdToShuffleDescriptorIndexMap.put(
-                    partitionId, partitionIdToShuffleDescriptorIndexMap.size());
+            Map<IntermediateResultPartitionID, Integer> partitionIdToShuffleDescriptorIndexMap,
```

Comment:
How to about to just pass in a `ConsumedPartitionGroup`?
It can better reflects the relationship of the context with the consumption group.